front/lib/api/data_sources.ts

```diff
@@ -73,8 +73,25 @@ export async function deleteDataSource(
     });
   }
 
-  const dustAPIProjectId = dataSource.dustAPIProjectId;
+  await dataSource.delete(auth, transaction);
```

Comment:
let's be explicit here and call softDelete?

Reply:
As per below: delete(hardDlete: false)

Reply:
I wanted to do this at first but in Typescript we can't change the parameter order or add required parameters when overriding 'delete' in 'B' because TypeScript requires method signatures in subclasses to be compatible with their superclasses. This ensures that 'B' instances can substitute 'A' instances without errors.

Will give it another shot with more templating!